### DISCUSSION TITLE
No device, no entities found[Bug]: 'Inverter Brand' - 'Inverter Model' Description of Error

### DISCUSSION THREAD

**Discussion by ececconi99** [wills106/homeassistant-solax-modbus#792] [Q&A]
### Describe the bug

I connected my home made Jumpy07 RS485 sniffer (https://github.com/Jumpy07/Solis---SolisCloud-and-Home-Assistant) between my Solis inverter and its datalogger today. 
I ran your HA Solax integration config, using MODBUS address 1 (a guess), Solis inverter, ELFIN/EW11 IP address and port. HA Solax said "success" and "initialising" but ended with no devices no entities. I emailed Solis support for the miodbus number (I could not find it on soliscloud or my inverter webpage).
Meanwhile, I tried various modbus numbers and worked my way down the SOLAX integration inverter list starting at Growatt. No dice until I tried Modbus 1 and inverter SRNE. 
Success! HA SOLAX int now reports 1 device and 24 entities.
So FYI, the Solis inverter config in the SOLAX integration does not seem to work, but using SRNE does! 
Do you have any thoughts on this?

### Integration Version

not known , where do I find this?

### Homeassistant core version

2024.2.5

### Inverter brand

Solis

### Plugin used

plugin_solis.py

### Serial prefix

6031020229030006  SOLIS S/N

### Connection Method

https://github.com/Jumpy07/Solis---SolisCloud-and-Home-Assistant

### Detailed Error Log

No errors reported

**Comment by ececconi99** [comment]
Update. After further troubleshooting with Jumpy07 on this "github issue"
https://github.com/Jumpy07/Solis---SolisCloud-and-Home-Assistant/issues/10 ,
I found that my Elfin EW11 RS485 to WIFI unit was not receiving or sending any data.  By trial and error I eventuallt realised that it seems the GC-1201 and EW11 manufacturers uses a different BUS naming convention so I had to swap the RS485 wires around, A-B and B-A, before I got any data throughput on the Elfin. So far so good.

However, in HA, using the "SolaX Inverter Modbus" integration (solis option) I get no devices no entities when I configure it. I see data traffic on the Elfin, so communication is there but no device. Any suggestions for further troubleshooting would be gratefully recieved.

**Reply by wills106** [reply]
I have moved this to the discussion section as it's more a connectivity question than a bug.

I would first recommend updating to 2024.03.1 there were changes and issues with 2024.2.x which are now resolved.
SRNE passes all the Entities regardless if you have communications or not for testing. So that's why it appears to be "working" You need to use the solis plugin.

**Comment by ececconi99** [comment]
> I would first recommend updating to 2024.03.1 there were changes and issues with 2024.2.x which are now resolved. SRNE passes all the Entities regardless if you have communications or not for testing. So that's why it appears to be "working" You need to use the solis plugin.
>>
Ok. I understand except for the update recommendation. Firstly, isn't 2024.03.1 to 2024.2.x a backward step? Secondly, how do I change versions? ATM, I just do as the HA system tells me and update whenever prompted to do so.

**Reply by wills106** [reply]
No, you need to go from the 2024.02.5 you on listed in your first post to the update from this morning which is 2024.03.1. 2024.03.1 fixes a left over issue from 2024.02.6 / 2025.02.7bX where the Integration wouldn't start communicating till you performed and additional restart following setting the Integration up.

Edit: Just realised you don't state the version. Only the HA Version.

Edit 2:
If you go to the HACS page in your HA, then Integrations and the SolaX Modbus one it's listed top left.
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/18155231/9533c6ab-486a-4996-ab79-aca2ab57afaa)

**Comment by ececconi99** [comment]
Thanks for the clarification.
I had updated this morning so I am up to date with v 2024.03.1 
I'm still getting "no device, no entities" when I run the config for this integration, despite seeing traffic on the RS485 bus on the ELFIN browser page. 
I do not know my inverter modbus address. I've looked at its browser page, on the Soliscloud device page and asked Solis support but no luck so far. I've tried 0, 1 and 2 so far but still get "no device, no entities"

**Comment by ececconi99** [comment]
re the modbus address, Solis support say"Please put your IP address in that section and try it and also you can try with the inverter serial number where there is second option of (serial)." They have also sent me a pdf document titled "RS485_MODBUS RTU Hybrid Inverter Protocol" It does not mention modbus address. 
We are clearly at cross purposes. 
You've told me I need to use the solis "inverter type" what about the other fields,
 Should I tick Parallel mode (master/slave)? TCP port? Modbus TCP?

**Comment by ececconi99** [comment]
Solis finally told me how to check or set the inverter slave modbus address number. I went up in the loft to check. It is set to "01"

I still would like you to tell me should I tick Parallel mode (master/slave)?
And confirm these settings, please; TCP port? Modbus TCP?
